### ISSUE TITLE
odo push and odo watch Fail to Update Non Static Content for Node 

### ISSUE THREAD

**Issue by danielhelfand** [openshift/odo#2272]
[kind/bug]

I used `odo` with the following components:
• [Node.js application](https://github.com/openshift-evangelists/Wild-West-Frontend)
• [Java application](https://github.com/openshift-evangelists/Wild-West-Backend)

After deploying and then linking the two components together as described under `How did you run odo exactly?`, making an update to non static content (i.e. anything under a `.js` file) fails to update after running `odo push` or while using `odo watch`. 

Updating simple things like text in html works as expected.

## What versions of software are you using?
- Operating System: macOD
- Output of `odo version`: Using build from #2265

### Java app

1. `odo create java backend --binary target/wildwest-1.0.jar`
2. `odo push`

### Node.js app

3. `odo create nodejs frontend`
4. `odo url create frontend`
5. `odo push`

### Linking

6. `odo link backend --component frontend --port 8080`

### Update Frontend
7. Updated the following [line of code](https://github.com/openshift-evangelists/Wild-West-Frontend/blob/7e35bbe9c272c4e43517d6e719395944a18ed453/assets/game.js#L100) by changing `'Score: 000'` to `'Score: 100'`
8. `odo push` (also tried by using `odo watch`)

## Actual behavior

`odo push` or `odo watch` completes successfully, but `'Score: 000'` remains for application when you access it by its url.

## Expected behavior

`'Score: 100' ` should be displayed

**Comment by danielhelfand** [comment]
@kadel Yes, apologies. I realize this app is probably not familiar to you, and I can certainly try using something simpler like `nodejs-ex` to see if this can be replicated.

As far as I can tell though, this score should still be updated as it should display upon loading the base url. Let me dig into the app itself and try another example, and I will follow up here.

**Comment by kadel** [comment]
I was playing with the app locally without odo and was able to reproduce it. It looks like it is browser caching, ctrl+shift+r in the browser shows the updated value.

But can't validate this on the cluster, as application fails on me before the request completes :-(
